File: src/my_cpp_pkg/src/add_two_ints_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"  

using std::placeholders::_1;
using std::placeholders::_2;

class AddTwoIntsServerNode: public rclcpp::Node {
public:
    AddTwoIntsServerNode(): Node("add_two_ints_server") {
        server_  = this->create_service<example_interfaces::srv::AddTwoInts>(
            "add_two_ints", std::bind(&AddTwoIntsServerNode::callback, this, _1, _2));
        RCLCPP_INFO(this->get_logger(), "Add Two Ints Server has been started.");

    }
private:
    rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;
    void callback(const std::shared_ptr<example_interfaces::srv::AddTwoInts::Request> request,
                  std::shared_ptr<example_interfaces::srv::AddTwoInts::Response> response) {
        response->sum = request->a + request->b;
        RCLCPP_INFO(this->get_logger(), "Incoming request\na: %ld" " b: %ld", request->a, request->b);
        RCLCPP_INFO(this->get_logger(), "Sending back response: %ld", response->sum);
    }

};

int main(int argsc, char **argsv) {
    rclcpp::init(argsc, argsv);
    auto node = std::make_shared<AddTwoIntsServerNode>();
    rclcpp::spin(node); 
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_server.py
#!/usr/bin/env python
import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts

class AddTwoIntsServer(Node):

    def __init__(self):
        super().__init__('add_two_ints_server')
        self.server = self.create_service(AddTwoInts, 'add_two_ints', self.add_two_ints_callback)
        self.get_logger().info('Add Two Ints Server has been started')

    def add_two_ints_callback(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info('Incoming request\na: %d b: %d' % (request.a, request.b))
        return response
    
def main(args=None):
    rclpy.init(args=args)
    add_two_ints_server = AddTwoIntsServer()
    rclpy.spin(add_two_ints_server)
    rclpy.shutdown()

if __name__ == '__main__':
    main()